packages/sdk/test/client.test.ts

```diff
@@ -1092,22 +1114,178 @@ describe('createEdition', () => {
+        mintConfigs: [mintConfig],
+        salt: SALT,
+      })
+      .catch((error) => {
+        expect(error).instanceOf(InvalidTimeValuesError)
+      })
+  })
+
+  it('throws if range mint cutoffTime === endTime', async () => {
+    const editionConfig = getGenericEditionConfig()
+
+    const mintConfig = getGenericRangeMintConfig()
+    mintConfig.cutoffTime = mintConfig.endTime
+
+    await client
+      .createEdition({
+        editionConfig,
+        mintConfigs: [mintConfig],
+        salt: SALT,
+      })
+      .catch((error) => {
+        expect(error).instanceOf(InvalidTimeValuesError)
+      })
+  })
+
+  it('throws if merkle mint merkleRoot is invalid', async () => {
+    const editionConfig = getGenericEditionConfig()
+
+    const startTime = now()
+    const endTime = startTime + 1
+
+    const mintConfig = {
+      mintType: 'MerkleDrop' as const,
+      minterAddress: merkleDropMinter.address,
+      price: PRICE,
+      merkleRoot: NULL_BYTES32,
+      startTime,
+      endTime,
+      maxMintable: 9,
+      maxMintablePerAccount: 1,
+      affiliateFeeBPS: 0,
+    }
+
+    await client
+      .createEdition({
+        editionConfig,
+        mintConfigs: [mintConfig],
+        salt: SALT,
+      })
+      .catch((error) => {
+        expect(error).instanceOf(InvalidMerkleRootError)
+      })
+
+    mintConfig.merkleRoot = '1x0000000000000000000000000000000000000000000000000000000000000000'
+
+    await client
+      .createEdition({
+        editionConfig,
+        mintConfigs: [mintConfig],
+        salt: SALT,
+      })
+      .catch((error) => {
+        expect(error).instanceOf(InvalidMerkleRootError)
+      })
+
+    mintConfig.merkleRoot = ''
+
+    client
+      .createEdition({
+        editionConfig,
+        mintConfigs: [mintConfig],
+        salt: SALT,
+      })
+      .catch((error) => {
+        expect(error).instanceOf(InvalidMerkleRootError)
+      })
+  })
 })
 
 describe('expectedEditionAddress', () => {
-  it('throws if provided deployerAddress is invalid', () => {
-    client.expectedEditionAddress({ deployer: '0x0', salt: '123' }).catch((error) => {
+  it('throws if provided deployerAddress is invalid', async () => {
+    await client.expectedEditionAddress({ deployer: '0x0', salt: '123' }).catch((error) => {
       expect(error).instanceOf(InvalidAddressError)
     })
   })
 
-  it('throws if provider not connected', () => {
-    client = SoundClient({ provider: new ethers.providers.JsonRpcProvider() })
+  it('throws if provider not connected', async () => {
+    client = SoundClient({
+      provider: new ethers.providers.JsonRpcProvider(),
+      soundCreatorAddress: soundCreator.address,
+    })
 
-    client
+    await client
       .expectedEditionAddress({ deployer: '0xbf9a1fad0cbd61cc8158ccb6e1e8e111707088bb', salt: '123' })
       .catch((error) => {
-        expect(error).instanceOf(MissingSignerOrProviderError)
+        expect(error.message).includes('could not detect network')
```

Comment:
Realized that we aren't awaiting these calls, which made the tests silently broken. I can move to a different PR if needed but seemed appropriate to include since they're testing validation.